### PR TITLE
Adds in the Tactical Deniability Implant for nukeops.

### PR DESCRIPTION
**Pull request by CRITAWAKETS**
## About The Pull Request

This PR adds in the Tactical Deniability Implant to the nuclear operative uplink for 6 TC. It's new version of the minibomb, implanted directly into your brain.
It disables softcrit and hardcrit, making it extremely powerful... For about 10.7 seconds (without stacked minibombs), before you realize what the increasing beeping is for, and proceed to explode. Even atropine can't save you from this one.

There is kind of a buff in that the delay of mini/macrobombs can no longer reach 30 seconds, since it just makes things a lot more interesting with a time cap (and also you'd need like 4 macrobombs to reach it so it mostly only impacts this new feature)

## Why It's Good For The Game

It's an interesting tradeoff where you can survive for a bit after critical condition, but whenever it triggers you've effectively sealed your own fate. Even if a helpful operative comes along to heal you, you WILL explode.

## Changelog

:cl:
add: The Syndicate has begun rolling out new Tactical Deniability Implants for their Nuclear Operative teams. It seems these implants are designed to make teams "fight harder" by "giving incentives for fighting to the bitter end", whatever they're talking about.
/:cl:

### CODE REVIEW

File: code/game/objects/items/implants/implant_explosive.dm [hunk 1]
```diff
@@ -149,6 +180,32 @@
 	explosion_heavy = 10 * MICROBOMB_EXPLOSION_HEAVY
 	explosion_devastate = 10 * MICROBOMB_EXPLOSION_DEVASTATE
 
+/obj/item/implant/explosive/deniability
+	name = "tactical deniability implant"
+	desc = "An enchanced version of the microbomb that directly plugs into the brain. No downsides, promise!"
```
Comment (Sealed101) [inline code/game/objects/items/implants/implant_explosive.dm:185]:
```suggestion
	desc = "An enhanced version of the microbomb that directly plugs into the brain. No downsides, promise!"
```

File: code/game/objects/items/implants/implant_explosive.dm [hunk 2]
```diff
@@ -120,19 +136,34 @@
...
+		while(delay > 1) //so we dont accidentally enter an infinite sleep
+			var/beep_volume = 35
+			playsound(loc, 'sound/items/timer.ogg', beep_volume, FALSE)
+			sleep(delay / 5)
+			delay -= delay / 5
+			beep_volume += 5
+
+
+///When called, just explodes
+/obj/item/implant/explosive/proc/explode()
```
Comment (Sealed101) [inline code/game/objects/items/implants/implant_explosive.dm:166]: might as well make use of this proc in `activate()`, L81

**Comment by TheObserver-sys** [comment]
This inspires the pure CEO mindset into any and all syndicate operatives. Encourages them to invest, to have a 10 year plan, demonstrates the fact they have the soul of an emperor when they live through an operation, that they have kept their divine light intact.

File: code/game/objects/items/implants/implant_explosive.dm [hunk 3]
```diff
@@ -73,25 +79,31 @@
...
+			var/obj/item/implant/explosive/other_implant = target_implant
+			if(other_implant.master_implant && master_implant) //we cant have two master implants at once
+				target.balloon_alert(target, "cannot fit implant!")
+				return FALSE
+			if(master_implant) //override the old implant and add in the old stats
+				explosion_devastate += other_implant.explosion_devastate
+				explosion_heavy += other_implant.explosion_heavy
+				explosion_light += other_implant.explosion_light
+				delay = min(delay + other_implant.delay, 30 SECONDS)
+				qdel(other_implant)
```
Comment (Time-Green) [inline code/game/objects/items/implants/implant_explosive.dm:98]: Could you put the explosion merging into it's own proc? Then you wont have to copypaste it in the next block

File: code/game/objects/items/implants/implant_explosive.dm [hunk 4]
```diff
@@ -120,19 +132,34 @@
...
+			sleep(delay * 0.25)
+			bomb_beeps_until_boom--
+			beep_volume += 5
+		explode()
+	else
+		addtimer(CALLBACK(src, PROC_REF(explode)), delay)
+		while(delay > 1) //so we dont accidentally enter an infinite sleep
+			var/beep_volume = 35
+			playsound(loc, 'sound/items/timer.ogg', beep_volume, FALSE)
+			sleep(delay / 5)
```
Comment (Time-Green) [inline code/game/objects/items/implants/implant_explosive.dm:156]:
```suggestion
			sleep(delay * 0.2)
```

File: code/game/objects/items/implants/implant_explosive.dm [hunk 5]
```diff
@@ -120,19 +132,34 @@
...
+			delay -= delay / 5
```
Comment (Time-Green) [inline code/game/objects/items/implants/implant_explosive.dm:157]:
```suggestion
			delay -= delay * 0.2
```
Multiplication is preferred when possible

File: code/game/objects/items/implants/implant_explosive.dm [hunk 6]
```diff
@@ -120,19 +132,34 @@
...
+			var/beep_volume = 35
+			playsound(loc, 'sound/items/timer.ogg', beep_volume, FALSE)
+			sleep(delay * 0.25)
+			bomb_beeps_until_boom--
+			beep_volume += 5
+		explode()
+	else
+		addtimer(CALLBACK(src, PROC_REF(explode)), delay)
+		while(delay > 1) //so we dont accidentally enter an infinite sleep
+			var/beep_volume = 35
```
Comment (Time-Green) [inline code/game/objects/items/implants/implant_explosive.dm:154]: I don't see the point in this var

File: code/game/objects/items/implants/implant_explosive.dm [hunk 7]
```diff
@@ -120,19 +132,34 @@
...
-		//for extra spice
-		var/beep_volume = 35
-		playsound(loc, 'sound/items/timer.ogg', beep_volume, FALSE)
-		sleep(delay * 0.25)
-		bomb_beeps_until_boom--
-		beep_volume += 5
+	if(!panic_beep_sound)
+		while(bomb_beeps_until_boom > 0)
+			//for extra spice
+			var/beep_volume = 35
```
Comment (Time-Green) [inline code/game/objects/items/implants/implant_explosive.dm:145]: Ditto

File: code/game/objects/items/implants/implant_explosive.dm [hunk 8]
```diff
@@ -120,19 +132,34 @@
...
+			playsound(loc, 'sound/items/timer.ogg', beep_volume, FALSE)
```
Comment (Time-Green) [inline code/game/objects/items/implants/implant_explosive.dm:146]:
```suggestion
			playsound(loc, 'sound/items/timer.ogg', vol = 35, vary = FALSE)
```

Use named params for readability

File: code/game/objects/items/implants/implant_explosive.dm [hunk 9]
```diff
@@ -120,19 +132,34 @@
...
+			sleep(delay * 0.25)
+			bomb_beeps_until_boom--
+			beep_volume += 5
+		explode()
+	else
+		addtimer(CALLBACK(src, PROC_REF(explode)), delay)
+		while(delay > 1) //so we dont accidentally enter an infinite sleep
+			var/beep_volume = 35
+			playsound(loc, 'sound/items/timer.ogg', beep_volume, FALSE)
```
Comment (Time-Green) [inline code/game/objects/items/implants/implant_explosive.dm:155]:
```suggestion
			playsound(loc, 'sound/items/timer.ogg', vol = 35, vary = FALSE)
```

Ditto

File: code/game/objects/items/implants/implant_explosive.dm [hunk 10]
```diff
@@ -149,6 +176,32 @@
 	explosion_heavy = 10 * MICROBOMB_EXPLOSION_HEAVY
 	explosion_devastate = 10 * MICROBOMB_EXPLOSION_DEVASTATE
 
+/obj/item/implant/explosive/deniability
```
Comment (Time-Green) [inline code/game/objects/items/implants/implant_explosive.dm:179]: Autodoc the type

File: code/game/objects/items/implants/implant_explosive.dm [hunk 11]
```diff
@@ -149,6 +176,32 @@
...
+	delay = 10 SECONDS
+	panic_beep_sound = TRUE
+	no_paralyze = TRUE
+	master_implant = TRUE
+
+/obj/item/implant/explosive/deniability/implant(mob/living/target, mob/user, silent = FALSE, force = FALSE)
+	. = ..()
+	if(.)
+		RegisterSignal(target, COMSIG_LIVING_HEALTH_UPDATE, PROC_REF(check_health))
+	target.add_traits(list(TRAIT_NOSOFTCRIT, TRAIT_NOHARDCRIT), IMPLANT_TRAIT)
```
Comment (Time-Green) [inline code/game/objects/items/implants/implant_explosive.dm:191]: I don't think this should be applying on failed implanting

File: code/game/objects/items/implants/implant_explosive.dm [hunk 12]
```diff
@@ -149,6 +176,32 @@
...
+
+/obj/item/implant/explosive/deniability/removed(mob/target, silent = FALSE, special = FALSE)
+	. = ..()
+	if(.)
+		UnregisterSignal(target, COMSIG_LIVING_HEALTH_UPDATE)
+	target.remove_traits(list(TRAIT_NOSOFTCRIT, TRAIT_NOHARDCRIT), IMPLANT_TRAIT)
```
Comment (Time-Green) [inline code/game/objects/items/implants/implant_explosive.dm:197]: Should this be getting removed on failed removal?

File: code/modules/uplink/uplink_items/nukeops.dm [hunk 13]
```diff
@@ -746,6 +746,15 @@
 	purchasable_from = UPLINK_NUKE_OPS
 	restricted = TRUE
 
+/datum/uplink_item/implants/macrobomb
```
Comment (Time-Green) [inline code/modules/uplink/uplink_items/nukeops.dm:749]: Macrobomb?

File: code/modules/uplink/uplink_items/nukeops.dm [hunk 14]
```diff
@@ -746,6 +746,15 @@
...
+	name = "Tactical Deniability Implant"
+	desc = "An implant injected into the brain, and later activated either manually or automatically upon entering critical condition. \
+			Prevents collapsing from critical condition, but explodes after a while."
+	item = /obj/item/storage/box/syndie_kit/imp_macrobomb
```
Comment (Time-Green) [inline code/modules/uplink/uplink_items/nukeops.dm:753]: Macrobomb?

**Review by Time-Green — CHANGES_REQUESTED** [review]
Could you just move this stuff into explode() aswell? 
image

Remind me that all the sleep-while code needs to be murdered

File: code/game/objects/items/implants/implant_explosive.dm (hunk 6 above)
Comment (Time-Green) [reply code/game/objects/items/implants/implant_explosive.dm:154]: Oh, I missed that

**Comment by CRITAWAKETS** [comment]
Hi, I've been sick a while, so it took a bit for me to get to fixing this up.

If anyone has any suggestion for the sleep-while code to be replaced with something which provides the same effect (a rapidly increasing beeping, dependant on delay) but isn't as poorly done, please do tell. I hate sleep() with a burning passion.

**Pull request merged**